app/labor/markdown_parser.rb

```diff
@@ -138,9 +138,12 @@ def escape_colon_emojis_in_codeblock(html)
   end
 
   def catch_xss_attempts(markdown)
-    bad_xss = ['src="data', "src='data", "src='&", 'src="&', "data:text/html"]
-    bad_xss.each do |xss_attempt|
-      raise ArgumentError, "Invalid markdown detected" if markdown.include?(xss_attempt)
+    bad_xss_regex = [
+      Regexp.new("src=[\"'](data|&)", Regexp::IGNORECASE),
+      Regexp.new("data:text/html[,;][\sa-z0-9]*", Regexp::IGNORECASE),
+    ]
+    bad_xss_regex.each do |xss_attempt|
```

Comment:
Instead of matching repeatedly, we can do the following:

```ruby
    if markdown.match?(Regexp.union(bad_xss_regex))
      raise ArgumentError, "Invalid markdown detected"
    end
```

You may want to do a bit more testing around it, but all current specs are green with this change in place, I tried locally.

Reply:
Nice, I'll resolve your comments this weekend. thanks!